src/contrib/cluster/Akka.Cluster.Sharding/ClusterSharding.cs

```diff
@@ -76,7 +76,7 @@ public ExtractorAdapter(IMessageExtractor underlying)
         {
             return message switch
             {
-                ShardingEnvelope se => se.Message,
+                ShardingEnvelope se => _underlying.EntityMessage(se.Message),
```

Comment:
Also call the underlying message extractor's `EntityMessage` method on the payload of the `ShardingEnvelope`, just in case that _also_ needs unwrapping. This is needed in order to make sure that messages still work properly when using tools like Akka.Cluster.Sharding.Delivery, which uses the `ShardingEnvelope` heavily.

Reply:
No because that's introducing side-effects the user is ultimately responsible for - we only care about the `ShardingEnvelope` specifically